### PR TITLE
Feature: Show error when a shortcut can't be created

### LINKED ISSUE #10987: Feature: "Can't create shortcut" modal
### What feature or improvement do you think would benefit Files?

Currently, there is no modal for a failed shortcut creation

An example of this is when you try to create a shortcut for an item in the root drive.
image

### Requirements

- Display a modal for a failed **Create Shortcut** operation
- Ask the user if they want to place the shortcut to the Desktop instead.

### Files Version

2.4.21

### Windows Version

22623.1095

### Comments

_No response_

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #10987 

**Details**
- Fixed a bug with shortcut's names created from modal (lines 82-86 `CreateShortcutDialogViewModel.cs`
- Added `Can't create shortcut` modal (I'm not sure if it should be displayed when `File/Folder` properties are edited)

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [x] Tested the changes for accessibility

**Screenshots (optional)**

### CODE REVIEW

File: src/Files.App/Strings/en-US/Resources.resw [hunk 1]
```diff
@@ -2877,4 +2877,10 @@
   <data name="ReleaseNotes" xml:space="preserve">
     <value>Release Notes</value>
   </data>
+  <data name="CannotCreateShortcutDialogTitle" xml:space="preserve">
+    <value>Files is not authorized to create a shortcut here.</value>
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:2881]:
```suggestion
    <value>Files can't create a shortcut here.</value>
```

**Comment by ferrariofilippo** [comment]
> I'm not sure if it should be displayed when File/Folder properties are edited

@yair100 what about this?

**Comment by yaira2** [comment]
What do you mean by this?

**Comment by ferrariofilippo** [comment]
At the moment, when the user opens `Properties` of a shortcut, he/she can edit its attributes. When this happens, Files tries to update the shortcut. If this is placed in a protected directory (such as `C:\`) nothing happens and the user doesn't get any error.

What I'm asking is whether the `Create on desktop` modal should be displayed in this specific case.
If the answer is no, should we inform the user that he/she needs to run Files as admin to successfully edit the shortcut?

**Comment by yaira2** [comment]
I didn't realize it can be worked around if the user runs as administrator. We should update the message accordingly. Could try something like this "This action requires administrator privileges", "Please run Files as administrator to create this shortcut".

**Comment by ferrariofilippo** [comment]
> I didn't realize it can be worked around if the user runs as administrator. We should update the message accordingly.

I think we should give the user the ability to create it on Desktop: if, for any reason , he can't run Files as admin, he should still be able to create that shortcut.

The modal might be like:
Title: `This action requires administrator privileges`
Message: `Please, run Files as administrator to create this shortcut or create it on Desktop`

**Comment by ferrariofilippo** [comment]
I have to inform you that I have temporarily removed the binding from `ItemDestinationPath`. Now this property is being updated manually. I did this because I noticed that the Binding was not working anymore. In fact, if you try to create a shortcut (even in stable version), you'll see that it doesn't work

File: src/Files.App/Strings/en-US/Resources.resw [hunk 2]
```diff
@@ -2877,6 +2877,12 @@
   <data name="ReleaseNotes" xml:space="preserve">
     <value>Release Notes</value>
   </data>
+  <data name="CannotCreateShortcutDialogTitle" xml:space="preserve">
+    <value>This action requires administrator privileges.</value>
+  </data>
+  <data name="CannotCreateShortcutDialogMessage" xml:space="preserve">
+    <value>Please, run Files as administrator or create it on Desktop</value>
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:2884]:
```suggestion
    <value>Please run Files as administrator or create the shortcut on the Desktop instead.</value>
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 3]
```diff
@@ -2871,6 +2871,12 @@
   <data name="ReleaseNotes" xml:space="preserve">
     <value>Release Notes</value>
   </data>
+  <data name="CannotCreateShortcutDialogTitle" xml:space="preserve">
+    <value>This action requires administrator privileges.</value>
+  </data>
+  <data name="CannotCreateShortcutDialogMessage" xml:space="preserve">
+    <value>Please run Files as administrator or create the shortcut on the Desktop instead.</value>
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:2878]:
```suggestion
    <value>Please run Files as administrator or create the shortcut somewhere else instead.</value>
```

**Review by yair100 — DISMISSED** [review]
LGTM

File: src/Files.App/Strings/en-US/Resources.resw (hunk 3 above)
Comment (ferrariofilippo) [reply src/Files.App/Strings/en-US/Resources.resw]: @yair100 at the moment if the creation of the shortcut fails and the user hits `Create` the shortcut will be placed on the Desktop

Comment (yaira2) [reply src/Files.App/Strings/en-US/Resources.resw]: Thanks! I edited the strings to clarify that.

**Pull request merged**